luxjs/services/pagination.js

```diff
@@ -0,0 +1,96 @@
+// Lux pagination module for controlling the flow of
+// repeat requests to the API.
+// It can return all data at an end point or offer
+// the next page on request for the relevant component
+
+angular.module('lux.pagination', ['lux.services'])
+    .factory('luxPaginationFactory', ['$lux', function($lux) {
+
+        /**
+        * LuxPagination constructor requires three args
+        * @param scope - the angular $scope of component's directive
+        * @param target {object} - containing name and url, e.g.
+        * {name: "groups_url", url: "http://127.0.0.1:6050"}
+        * @param recursive {boolean}- set to true if you want to recursively
+        * request all data from the endpoint
+        */
+
+        function LuxPagination(scope, target, recursive) {
+            this.scope = scope;
+            this.target = target;
+            this.orgUrl = this.target.url;
+            this.api = $lux.api(this.target);
+
+            if (recursive === true) this.recursive = true;
+        }
+
+        LuxPagination.prototype.getData = function(params, cb) {
+            // getData runs $lux.api.get() followed by the component's
+            // callback on the returned data or error.
+            // it's up to the component to handle the error.
+
+            this.params = params ? params : null;
+            if (cb) this.cb = cb;
+
+            this.api.get(null, params).then(function(data) {
+
+                if (data.error) return cb(data);
```

Comment:
I've added the error callback which will pass the data (containing the error and message) into the callback so the component can choose how to deal with the error.

I'll talk to you about notification callbacks when we're both in the office.